[FEATURE] Avoir une navigation principale pour Pix Pro (PIX-1368).

MelanieMEB opened the pull request:
## :unicorn: Problème
Le site pro.pix va avoir sa propre navigation principale et n'utilisera plus les sub-nav.

## :robot: Solution
- Coté Prismic : ajout d'un attribut `navigation_for` avec comme valeur pix-site/pix-pro sur le type `Main Navigation`
- Rendre le type `Main Navigation` répétable
- Ajouter une méthode pour récupérer tous les documents d'un même type
- Dans `NavigationSliceZone`, regrouper (par Lodash qui a été ajouté) les navigations par site utilisé, et prendre la première navigation proposé pour le site actuel
- Si pro.pix n'a pas de navigation (comme actuellement sur la prod), prendre celle de pix-site
- Si pro.pix a une navigation, retirer la sub-nav qui devient inutile (aussi dans le burger).

## :rainbow: Remarques
- Ajout de Lodash
- Ajout de tests 😸 

## :sparkles: Review App
https://site-pr190.review.pix.fr/
https://pro-pr190.review.pix.fr/

MelanieMEB committed:
Add findByType method to find all main_navigation typed document

MelanieMEB committed:
Use findByType method to get the main Navigation

MelanieMEB committed:
Add Lodash to the project

MelanieMEB committed:
Add usedMainNavigation in NavigationSliceZone to choose between pix-pro or pix-site nav

MelanieMEB committed:
Hide sub nav in Pix Pro when the main navigation is already the pix-pro navigation

alexandrecoin commented on store/index.js:
```diff
@@ -51,7 +51,7 @@ export const mutations = {
   },
   updateMainNavigation(state, navigations) {
-    state.mainNavigation = { ...navigations }
+    state.mainNavigation = navigations
```
**Suggestion :**  Il est recommandé par la doc de Vuex de créer un nouvel objet (ou array) lors de la mutation des données. [Lien](https://vuex.vuejs.org/fr/guide/mutations.html#les-mutations-suivent-les-regles-de-reactivite-de-vue)

```suggestion
    state.mainNavigation = [...navigations]
```

MelanieMEB replied:
Corrigé !

alexandrecoin commented on components/NavigationSliceZone.vue:
```diff
@@ -43,6 +43,7 @@
 
 <script>
 import { mapState } from 'vuex'
+import { groupBy } from 'lodash'
```
**Question :** Serait-il envisageable de passer outre l'ajout de Lodash vu le peu de fonctions de la librairie nécessaires au projet ?

celineung replied:
Je me pose aussi la même question.
Mais vu que le projet commence à grossir, à quel moment se dirait-on que c'est le bon moment de le rajouter?

Laurentlcs replied:
Vu que c'est importé par méthode ce n'est pas trop lourd, et la méthode groupBy en JS natif est assez verbeuse..
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_groupby
Pour cette méthode je pense qu'utiliser Lodash n'est pas un souci. 

Ou alors créer une méthode groupBy à partir du code natif JS.

MelanieMEB replied:
Je propose de laisser, le site allant devenir de + en + complexe, ça sera utile de l'avoir sous la main.
Mais il faut qu'on garde l'habitude d'import que ce dont on a besoin et pas tout lodash :)

alexandrecoin commented on store/index.js:
```diff
@@ -51,7 +51,7 @@ export const mutations = {
   },
   updateMainNavigation(state, navigations) {
-    state.mainNavigation = { ...navigations }
+    state.mainNavigation = navigations
```
**Nitpick** : Remplacer state.mainNavigation et ses occurrences vu qu'il s'agit désormais d'un tableau de navigations.
```suggestion
    state.mainNavigations = navigations
```

MelanieMEB replied:
Corrigé !

MelanieMEB committed:
[Clean] remplace mainNavigation by mainNavigations

Replace the object for the state : https://vuex.vuejs.org/fr/guide/mutations.html#les-mutations-suivent-les-regles-de-reactivite-de-vue

MelanieMEB commented on components/NavigationSliceZone.vue:
```diff
@@ -65,6 +69,11 @@ export default {
     },
+
+    showSubNav() {
+      return this.usedMainNavigation.data.navigation_for !== 'pix-pro'
```
```suggestion
      return this.usedMainNavigation.data.navigation_for === 'pix-site'
```

MelanieMEB replied:
Modifié

celineung commented on components/NavigationSliceZone.vue:
```diff
@@ -65,6 +66,14 @@ export default {
+    showSubNav() {
+      return (
+        this.isPixPro &&
+        this.usedMainNavigation.data.navigation_for === 'pix-site'
```
**question**: est-ce que cette deuxième condition est nécessaire? Car `this.usedMainNavigation` se définit déjà en fonction de `this.isPixPro`.

MelanieMEB replied:
Ici, c'est le cas où on est sur PixPro et qu'on utiliser la nav de pix-site (quand on sera encore entre les deux), donc ce n'est pas exactement le même cas (et normalement c'est un cas voué à disparaitre

celineung commented:
Comment teste-on la sous-navigation? Je ne parviens pas à accéder à la page pix-pro.

MelanieMEB commented:
> Comment teste-on la sous-navigation? Je ne parviens pas à accéder à la page pix-pro.

Si tu vas sur la review app Pro : la navigation ne dois pas changer.
Sinon, sur Prismic, tu as la Main navigation pix-pro que tu peux preview (et là, tu verras la nouvelle navigation, et sans la subnav)

pix-service commented:
I'm deploying this PR to these urls:

- Pix Site (fr): https://site-pr190.review.pix.fr
- Pix Site (org): https://site-pr190.review.pix.org
- Pix Pro (fr): https://pro-pr190.review.pix.fr
- Pix Pro (org): https://pro-pr190.review.pix.org

 Please check it out!
